### PR TITLE
FEATURE: destroy old hidden queries

### PR DESCRIPTION
**Pull request by techAPJ**
This commit adds a scheduled job to hard delete queries that are hidden and

- were last run more than 7 days ago
- were updated more than 7 days ago

### CODE REVIEW

File: app/jobs/scheduled/delete_hidden_queries.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+module Jobs
+  class DeleteHiddenQueries < ::Jobs::Scheduled
+    every 7.days
+
+    def execute(args)
+      return unless SiteSetting.data_explorer_enabled
+
+      DataExplorer::Query.where(hidden: true).each do |query|
+        if query.last_run_at < 7.days.ago && query.updated_at < 7.days.ago
```
Comment (davidtaylorhq) [inline app/jobs/scheduled/delete_hidden_queries.rb:11]: Can `last_run_at` be `nil`?

File: app/jobs/scheduled/delete_hidden_queries.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Jobs
+  class DeleteHiddenQueries < ::Jobs::Scheduled
+    every 7.days
+
+    def execute(args)
+      return unless SiteSetting.data_explorer_enabled
+
+      DataExplorer::Query.where(hidden: true).each do |query|
```
Comment (davidtaylorhq) [inline app/jobs/scheduled/delete_hidden_queries.rb:10]: Some sites have stacked up 10,000+ hidden queries, so we should probably do this in batches... Or alternatively do the `last_run_at` and `updated_at` checks in SQL, and use `.delete_all`

**Review by davidtaylorhq — COMMENTED** [review]
Looks good. What happens if you press delete on one of the pre-seeded queries in the UI? Does [the deletion](https://github.com/discourse/discourse-data-explorer/blob/master/app/controllers/data_explorer/query_controller.rb#L102) persist? Or does the query re-appear after 7 days? I think we might need to skip destroying rows with ID < 0 🤔

**Comment by techAPJ** [comment]
> Looks good. What happens if you press delete on one of the pre-seeded queries in the UI? Does [the deletion](https://github.com/discourse/discourse-data-explorer/blob/master/app/controllers/data_explorer/query_controller.rb#L102) persist? Or does the query re-appear after 7 days? I think we might need to skip destroying rows with ID < 0 🤔

The pre-seeded queries can not be deleted from UI because we hide the delete button for them.

File: app/jobs/scheduled/delete_hidden_queries.rb (hunk 2 above)
Comment (techAPJ) [reply app/jobs/scheduled/delete_hidden_queries.rb:10]: Good idea. Will make it so.

File: app/jobs/scheduled/delete_hidden_queries.rb (hunk 1 above)
Comment (techAPJ) [reply app/jobs/scheduled/delete_hidden_queries.rb:11]: Yes, let me check for the presence of `last_run_at`.

**Comment by davidtaylorhq** [comment]
> The pre-seeded queries can not be deleted from UI because we hide the delete button for them.

Hmm... I think it is supposed to work. The reason the 'hidden' feature was added was [to delete system queries](https://meta.discourse.org/t/how-to-delete-system-queries-in-data-explorer/147524/8?u=david). On localhost I see the delete button for default queries. (and I can also edit them... which seems unintended... but that's a different problem)

**Comment by techAPJ** [comment]
@davidtaylorhq Updated the PR per your feedback.

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**